### PR TITLE
Kixer ads: Fire viewability when ad becomes visible

### PR DESCRIPTION
**Pull request by jvelis**
We would like to improve viewability numbers by using the position in the viewport to fire the view.
- Using window.context.observeIntersection API to detect the position.

### CODE REVIEW

**Comment by erwinmombay** [comment]
@ampprojectbot retry!

**Comment by jvelis** [comment]
I signed it!

**Comment by jvelis** [comment]
@googlebot I signed it!

File: ads/kixer.js [hunk 1]
```diff
@@ -45,6 +46,17 @@ export function kixer(global, data) {
   };
   d.addEventListener('load', kxload, false);
 
+  const unlisten = global.context.observeIntersection(function(changes) {
+    changes.forEach(function(c) {
+      if (c.intersectionRect.height > 0) {
+        if (typeof __kx_viewability.process_locked === 'function') {
```
Comment (lannka) [inline ads/kixer.js]: to handle race condition, you might want to do something like:

```js
const loadPromise = loadScriptPromise(global, 'https://cdn.kixer.com/ad/load.js')
   .then(() => {
      ...
    });

const unlisten = global.context.observeIntersection(function(changes) {
  for (...) {
      if (c.intersectionRect.height > 0) {
        loadPromise.then(() => {

        );
        break;
      }
  }
});

```

File: ads/kixer.js [hunk 2]
```diff
@@ -45,6 +46,17 @@ export function kixer(global, data) {
...
+          __kx_viewability.process_locked(data.adslot);
+        }
+        unlisten();
```
Comment (jvelis) [inline ads/kixer.js]: Hi Lannka, thanks for the feedback. 
This should be called only once now, although it wouldn't affect if we get multiple calls.

File: ads/kixer.js (hunk 1 above)
Comment (jvelis) [inline ads/kixer.js]: Makes sense. I tried the loadPromise suggestion but we use our own 'load' event to know when the ad has loaded, so I moved the viewability check to the listener.

**Comment by jvelis** [comment]
@googlebot I signed it!

**Comment by jvelis** [comment]
@googlebot I confirm.

**Comment by jvelis** [comment]
Closing due to rebase issue.

New PR is #8289

**Pull request closed without merging**